### PR TITLE
Arbitrary ORTB request configuration changes

### PR DESCRIPTION
**Pull request by OlenaPostindustria**
#1017

### CODE REVIEW

File: PrebidMobile/PrebidMobileRendering/Networking/Parameters/ArbitraryORTBHelper.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+}
+
+class ArbitraryGlobalORTBHelper: ArbitraryORTBHelperProtocol {
+    
+    private let ortb: String?
+    
+    struct ProtectedFields {
+        
+        static var deviceProps: [String] {
```
[the remark this answers is not in the document]
Comment (OlenaPostindustria) [reply PrebidMobile/PrebidMobileRendering/Networking/Parameters/ArbitraryORTBHelper.swift:48]: Yes, it seems like a more memory-efficient solution.

File: PrebidMobileTests/RenderingTests/Tests/ParameterBuilderTests/ArbitraryORTBHelperTests.swift [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            "key1": "value1",
+            "key2":
+        """
+        
+        let result = ArbitraryImpORTBHelper(ortb: invalidJSON).getValidatedORTBDict()
+        
+        XCTAssertNil(result)
+    }
+    
+    func testGlobalORTBHelper_ValidJSON_RemoveProtectedFields() {
```
[the remark this answers is not in the document]
Comment (OlenaPostindustria) [reply PrebidMobileTests/RenderingTests/Tests/ParameterBuilderTests/ArbitraryORTBHelperTests.swift:48]: Already added some tests in [ArbitraryORTBServiceTests](https://github.com/prebid/prebid-mobile-ios/blob/d194a009a3705e3017fedcfbd9a81f7c8224447a/PrebidMobileTests/RenderingTests/Tests/ParameterBuilderTests/ArbitraryORTBServiceTests.swift).

**Review by YuriyVelichkoPI — APPROVED** [review]
LGTM

File: PrebidMobile/ConfigurationAndTargeting/Targeting.swift [hunk 3]
```diff
@@ -303,6 +303,20 @@ public class Targeting: NSObject {
     
     // MARK: - Public Methods
     
+    // MARK: Arbitrary ORTB Configuration
+    
+    /// Sets the global-level OpenRTB configuration string.
+    ///
+    /// - Parameter ortbObject: The global-level OpenRTB configuration string to set. Can be `nil` to clear the configuration.
+    public func setGlobalORTBConfig(_ ortbConfig: String?) {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/ConfigurationAndTargeting/Targeting.swift:311]: Please add an example of how this method is used in the demo apps. 

Let's set: 
- `ext.myext.test = 1` 
- `displaymanager` = Google
- `displaymanagerver` = $CurrentGMASDKVersion

File: PrebidMobile/AdUnits/MultiformatAdUnit/PrebidRequest.swift [hunk 4]
```diff
@@ -64,6 +64,20 @@ public class PrebidRequest: NSObject {
         self.gpid = gpid
     }
     
+    // MARK: Arbitrary ORTB Configuration
+    
+    /// Sets the impression-level OpenRTB configuration string for the ad unit.
+    ///
+    /// - Parameter ortbObject: The  impression-level OpenRTB configuration string to set. Can be `nil` to clear the configuration.
+    public func setImpressionORTBConfig(_ ortbConfig: String?) {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/AdUnits/MultiformatAdUnit/PrebidRequest.swift:72]: Please add an example of how this method is used in the demo apps. 

Let's set: 
- `imp.bidfloor=10` 
- `imp.banner.battr` = [1,2,3,4]

**Review by YuriyVelichkoPI — CHANGES_REQUESTED** [review]

File: PrebidMobile/AdUnits/AdUnit.swift [hunk 5]
```diff
@@ -409,23 +409,21 @@ public class AdUnit: NSObject, DispatcherDelegate {
...
+    // MARK: Arbitrary ORTB Configuration
     
-    /// Sets the OpenRTB configuration string for the ad unit.
+    /// Sets the impression-level OpenRTB configuration string for the ad unit.
     ///
-    /// - Parameter ortbObject: The OpenRTB configuration string to set. Can be `nil` to clear the configuration.
-    public func setOrtbConfig(_ ortbObject: String?) {
-        adUnitConfig.ortbConfig = ortbObject
+    /// - Parameter ortbObject: The impression-level OpenRTB configuration string to set. Can be `nil` to clear the configuration.
+    public func setImpressionORTBConfig(_ ortbConfig: String?) {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/AdUnits/AdUnit.swift:417]: Please, rename to `setImpORTBConfig`

**Review by YuriyVelichkoPI — CHANGES_REQUESTED** [review]

**Review by YuriyVelichkoPI — APPROVED** [review]

**Pull request merged**